### PR TITLE
Qwen3 model support

### PR DESCRIPTION
**Pull request by vaibhavjindal**
## Summary

This pull request introduces support for the Qwen3 model within the Liger-Kernel framework.
Fixes: https://github.com/linkedin/Liger-Kernel/issues/690

## Testing Done

A simple check:
```
from transformers import Qwen3ForCausalLM
from liger_kernel.transformers.monkey_patch import apply_liger_kernel_to_qwen3

apply_liger_kernel_to_qwen3()
model = Qwen3ForCausalLM.from_pretrained("Qwen/Qwen3-8B")

print(model)
```
gives the following output:
Screenshot 2025-05-02 at 12 15 14 AM

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by vaibhavjindal** [comment]
https://github.com/linkedin/Liger-Kernel/issues/690

File: src/liger_kernel/transformers/model/qwen3.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            lm_head_weight=self.lm_head.weight,
+            labels=labels,
+            shift_labels=shift_labels,
+            hidden_size=self.config.hidden_size,
+            **kwargs,
+        )
+
+    else:  # if in inference mode materialize logits
+        slice_indices = slice(-logits_to_keep, None) if isinstance(logits_to_keep, int) else logits_to_keep
+        logits = self.lm_head(hidden_states[:, slice_indices, :])
```
Comment (Tcc0403) [inline src/liger_kernel/transformers/model/qwen3.py:101]: I just realized that we should make it work on both training mode and inference mode. 
Let's address this issue in qwen3. Fixes for other models can be another PR.
 
More details: #694.

**Comment by shivam15s** [comment]
LGTM + let's use `logits_to_keep` as @Tcc0403 suggested

**Comment by vaibhavjindal** [comment]
@shivam15s @Tcc0403 fixed the slicing issue for training!

**Review by shivam15s — APPROVED** [review]
lgtm
